src/NuGet.Core/NuGet.Build.Tasks/CheckForDuplicateNuGetItemsTask.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.ProjectModel;
+
+
+namespace NuGet.Build.Tasks
+{
+    public class CheckForDuplicateNuGetItemsTask : Task
```

Comment:
I think this task should implement    https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.framework.icancelabletask interface so that it supports cancellations.
I remember seeing an issue on NuGet/Home where an issue was created to add support for 'GetRestoreSettings' task. Please let me know if I'm missing anything.

Reply:
Maybe that issue can add cancellable interface to all the tasks?

There's a big difference between GetRestoreSettings and this task though.  
GetRestoreSettings does a lot of IO which can be slow, while this is just simple compute, so I don't think it'd do much.

Reply:
> existing code does it this way, so if we want to change the design patterns we use, do it in another issue where we change the whole repo

I think that's a reach. It's more of, on first look, this task makes no sense as ICancellable, so that other task can spend the time investigating and change it if they really think otherwise. 

Resolving the comment thread as this task is simply not suitable for adding cancellation.